Configure External Slurm

aabills opened the pull request:
We need to configure spack to use our own slurm. Buildable should be false, since there is no reason for users to be installing their own slurm, and it's likely to lead to user confusion. see https://github.com/ArjunaCluster/ArjunaUsers/issues/53

aabills committed:
use our own slurm

awadell1 reviewed (approved):
Buildable false is probably not required here, but as users can override it's probably a good default.

Merge Actions: Pull and update overlays

awadell1 committed:
fix: Correct slurm spec

Fixed version to match slurm's format on github
Added sysconfdir spec (Unlikly to be important)

Leaving other varients at defaults
